### PR TITLE
Legacy parser for V9-V13

### PR DESCRIPTION
**Pull request by justkawal**
## Any follow ups?
#257 

## Description
- Adds Set implementation
- Resolves Indexed Enum in the Registry
- Adds implementation for Legacy Parser to create `event` and `extrinsics` lookup.

## Testing
- `dart test`

### CODE REVIEW

File: packages/polkadart_scale_codec/lib/core/registry.dart [hunk 1]
```diff
@@ -117,6 +117,8 @@ class Registry {
               return _parseResult(customJson, key, value);
             case 'BitVec':
               return _parseBitSequence(customJson, key, value);
+            case 'DoNotConstruct':
+              return NullCodec.instance;
```
Comment (Lohann) [inline packages/polkadart_scale_codec/lib/core/registry.dart:121]: ? this is for internal use only?

File: packages/polkadart_scale_codec/lib/primitives/enum.dart [hunk 2]
```diff
@@ -1,11 +1,12 @@
 part of primitives;
 
 class SimpleEnumCodec<A> with Codec<A> {
-  final List<A> list;
+  final List<A?> list;
```
Comment (Lohann) [inline packages/polkadart_scale_codec/lib/primitives/enum.dart:4]: Why not using a map instead an array?

Comment (justkawal) [reply packages/polkadart_scale_codec/lib/primitives/enum.dart:4]:
```
 final output = HexOutput();
 final codec = SimpleEnumCodec(['a', 'b', 'c']);
 codec.encodeTo('b', output);
 expect(output.toString(), '0x01');
```

It is useful to automatically pick the index

Also don't have to do: ` codec.encodeTo(MapEntry('b', null), output);`

Can avoid writing null again and again.

It is changed to A? in the PR: #186 .

File: packages/polkadart_scale_codec/lib/core/registry.dart (hunk 1 above)
Comment (justkawal) [reply packages/polkadart_scale_codec/lib/core/registry.dart:121]: Yes, We will not export it and hide the type in the exporter file.

File: packages/polkadart_scale_codec/lib/primitives/enum.dart (hunk 2 above)
Comment (justkawal) [reply packages/polkadart_scale_codec/lib/primitives/enum.dart]: Resolved by forwarding changes to child PRs.

**Pull request closed without merging**